src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -117,6 +118,14 @@ internal class SqlServerSearchService : SearchService
             _schemaInformation = schemaInformation;
             _requestContextAccessor = requestContextAccessor;
             _compressedRawResourceConverter = compressedRawResourceConverter;
+
+            if (_resourceSearchParamStats == null)
+            {
+                lock (_locker)
+                {
+                    _resourceSearchParamStats ??= new ResourceSearchParamStats(_sqlRetryService, _logger, _model);
```

Comment:
Should it check that the stats are still null after acquiring the lock?

Reply:
Not sure what conditions could lead to that. As long as constructor succeeds it is not null...

Reply:
I believe the situation RB is suggesting is: 
- Two threads reach line 123 at the same time. 
- First one of the two threads reach line 125 first. 
- Second thread waits for the lock.
- First thread finishes the work.
- Second thread creates another instance of ResourceSearchParamStats.

Reply:
I missed you were using a null check assignment operator.

Reply:
Yes... _resourceSearchParamStats ??=